openquake/hazardlib/gsim/germany_2018.py

```diff
@@ -0,0 +1,182 @@
+#
+# OpenQuake is free software: you can redistribute it and/or modify it
+# under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# OpenQuake is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with OpenQuake. If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Implements the GMPE adjustment used for the Cauzzi et al. (2014) and
+Derras et al. (2014) model, as used in the 2018 German National Seismic Hazard
+Model of Gruenthal et al. (2018)
+
+"Gruenthal, G., Strometer, D., Bosse, C., Cotton, F. and Bindi, D. (2018)
+The probabilistic seismic hazard assessment of Germany - version 2016,
+considering the range or epistemic uncertainties and aleatory variability."
+Bulletin of Earthquake Engineering, 16(10), 4339-4395
+
+Module Exports :class: `CauzziEtAl2014RhypoGermany`,
+                       `DerrasEtAl2014RhypoGermany`
+"""
+import numpy as np
+from scipy.constants import g
+from openquake.hazardlib.imt import PGA, SA
+from openquake.hazardlib.gsim.cauzzi_2014 import CauzziEtAl2014
+from openquake.hazardlib.gsim.derras_2014 import DerrasEtAl2014
+
+
+def rhypo_to_rrup(rhypo, mag):
+    """
+    Converts hypocentral distance to an equivalent rupture distance
+    dependent on the magnitude
+    """
+    rrup = rhypo - (0.7108 + 2.496E-6 * (mag ** 7.982))
+    rrup[rrup < 3.0] = 3.0
+    return rrup
+
+
+def rhypo_to_rjb(rhypo, mag):
+    """
+    Converts hypocentral distance to an equivalent Joyner-Boore distance
+    dependent on the magnitude
+    """
+    epsilon = rhypo - (4.853 + 1.347E-6 * (mag ** 8.163))
+    rjb = np.zeros_like(rhypo)
+    idx = epsilon >= 3.
+    rjb[idx] = np.sqrt((epsilon[idx] ** 2.) - 9.0)
+    rjb[rjb < 0.0] = 0.0
+    return rjb
+
+
+# Cauzzi et al. 2014 - Converted from Rhypo
+class CauzziEtAl2014RhypoGermany(CauzziEtAl2014):
+    """
+    Implements the Cauzzi et al. (2015) GMPE applying the rhypo to rrup
+    adjustment factor adopted for Germany
+    """
+    REQUIRES_DISTANCES = set(("rhypo", ))
+
+    def __init__(self, adjustment_factor="1.0"):
+        super().__init__()
```

Comment:
Why the adjustement is a string here and a float a few classes below? I would prefer a string if it works.